src/chain/chain_accumulator.rs

```diff
@@ -9,8 +9,65 @@
 use super::{AccumulatingEvent, NetworkEvent, Proof, ProofSet, SectionInfoSigPayload};
 use crate::id::PublicId;
 use log::LogLevel;
-use std::collections::{BTreeMap, BTreeSet};
-use std::mem;
+use std::collections::{BTreeMap, BTreeSet, VecDeque};
+use std::{mem, rc::Rc};
+
+/// The threshold (number of unvoted votes) a node to be considered as unresponsive.
+pub const UNRESPONSIVE_THRESHOLD: usize = 48;
+/// The period (X consensued observations) after which node be considered as unresponsive.
+pub const UNRESPONSIVE_WINDOW: usize = 64;
+
+#[derive(Default)]
+struct VoteStatuses {
+    tracked_events: VecDeque<Rc<AccumulatingEvent>>,
+    unvoted: BTreeMap<PublicId, BTreeSet<Rc<AccumulatingEvent>>>,
+}
+
+impl VoteStatuses {
+    fn add_expectation(
+        &mut self,
+        event: Rc<AccumulatingEvent>,
```

Comment:
ideally, this implementation detail would not leak in the interface, i.e pass an AccumulatingEvent and build the Rc from within. Depends on the other comment though.